example/generator/odp_generator.c

```diff
@@ -784,10 +811,33 @@ static void print_pkts(int thr, thread_args_t *thr_args,
 	unsigned i;
 	size_t offset;
 	char msg[1024];
+	interface_t *itfs, *itf;
+
+	itfs = thr_args->rx.ifs;
 
 	for (i = 0; i < len; ++i) {
 		pkt = pkt_tbl[i];
 
+		itf = &itfs[odp_pktio_index(odp_packet_input(pkt))];
+
+		if (odp_packet_has_ipv4(pkt)) {
+			if (itf->config.pktin.bit.ipv4_chksum) {
+				if (odp_packet_has_l3_error(pkt))
+					printf("HW detected L3 error\n");
+			}
+		}
+
+		if (odp_packet_has_udp(pkt)) {
+			if (itf->config.pktin.bit.udp_chksum) {
+				if (odp_packet_has_l4_error(pkt))
+					printf("HW detected L4 error\n");
+			}
+		}
+
+		/* Drop packets with errors */
+		if (odp_unlikely(odp_packet_has_error(pkt)))
```

Comment:
that has to be the first check.

Reply:
No -  we need to print csum errors first.
This part was significantly changed in api-next (csum checks use different/ new API) and it makes no sense to optimize it for the old (master) code. After integration in api-next, this part will be  reworked  to use less parser flags (reduce parsing level).
For example, removing L4 parsing and locating interface is bringing an extra 1 mpps.

Reply:
ok

Reply:
Checksum errors will result in `odp_packet_has_error()` being set as well, so these checks can be done only if the summary packet error predicate is set, avoiding unnecessary checks for known good packets.

Reply:
This csum check is done with newer API in API-NEXT (odp_packet_l3_chksum_status()). No sense to optimize this part for this older implementation